pkg/admissioncontroller/webhook/auth/seed/graph/eventhandler_seed.go

```diff
@@ -97,8 +99,18 @@ func (g *graph) handleSeedCreateOrUpdate(seed *gardencorev1beta1.Seed) {
 	g.addEdge(leaseVertex, seedVertex)
 
 	if seed.Spec.Backup != nil {
-		secretVertex := g.getOrCreateVertex(VertexTypeSecret, seed.Spec.Backup.SecretRef.Namespace, seed.Spec.Backup.SecretRef.Name)
-		g.addEdge(secretVertex, seedVertex)
+		var (
+			namespace = seed.Spec.Backup.CredentialsRef.Namespace
+			name      = seed.Spec.Backup.CredentialsRef.Name
+			vertex    *vertex
+		)
+		if seed.Spec.Backup.CredentialsRef.APIVersion == securityv1alpha1.SchemeGroupVersion.String() &&
+			seed.Spec.Backup.CredentialsRef.Kind == "WorkloadIdentity" {
+			vertex = g.getOrCreateVertex(VertexTypeWorkloadIdentity, namespace, name)
+		} else {
```

Comment:
`else if` to be sure it's a `Secret`?

Reply:
The API validation does not allow anything else other than Secret and WorkloadIdentity, but in case of new credentials implementations the proposed change should be helpful.

Reply:
Yep :)